library/src/main/java/org/eclipse/uprotocol/ULink.java

```diff
@@ -0,0 +1,779 @@
+     *
+     * @return A {@link UEntity}.
+     */
+    public @NonNull UEntity getEntity() {
+        return mClientUri.getEntity();
+    }
+
+    /**
+     * Get a URI of a client who is holding this instance.
+     *
+     * @return A {@link UUri}.
+     */
+    public @NonNull UUri getClientUri() {
+        return mClientUri;
+    }
+
+    @VisibleForTesting
+    @NonNull String getTag() {
+        return mTag;
+    }
+
+    @VisibleForTesting
+    void setLoggable(int level) {
+        mVerboseLoggable = level <= Log.VERBOSE;
+    }
+
+    @VisibleForTesting
+    ConnectionCallback getConnectionCallback() {
+        return mConnectionCallback;
+    }
+
+    @VisibleForTesting
+    UListener getListener() {
+        return mListener;
+    }
+
+    private static @NonNull UMessage buildMessage(UUri source, UPayload payload, UAttributes attributes) {
+        final UMessage.Builder builder = UMessage.newBuilder();
+        if (source != null) {
+            builder.setSource(source);
+        }
+        if (payload != null) {
+            builder.setPayload(payload);
+        }
+        if (attributes != null) {
+            builder.setAttributes(attributes);
+        }
+        return builder.build();
+    }
+
+    @Override
+    public @NonNull UStatus send(@NonNull UUri source, @NonNull UPayload payload, @NonNull UAttributes attributes) {
+        return mClient.send(buildMessage(source, payload, attributes));
+    }
+
+    @Override
+    public @NonNull UStatus send(@NonNull UMessage message) {
+        return mClient.send(message);
+    }
+
+    @Override
+    public @NonNull UStatus registerListener(@NonNull UUri topic, @NonNull UListener listener) {
+        try {
+            checkArgument(!isEmpty(topic), "Topic is empty");
+            checkNotNull(listener, "Listener is null");
+            synchronized (mRegistrationLock) {
+                Set<UListener> listeners = mListeners.get(topic);
+                if (listeners == null) {
+                    listeners = new HashSet<>();
+                }
+                if (listeners.isEmpty()) {
+                    final UStatus status = mClient.enableDispatching(topic);
+                    if (!isOk(status)) {
+                        return status;
+                    }
+                    mListeners.put(topic, listeners);
+                }
+                if (listeners.add(listener) && listeners.size() > 1) {
+                    mCallbackExecutor.execute(() -> {
+                        final UMessage event = mClient.getLastMessage(topic);
+                        if (event != null) {
+                            listener.onReceive(event);
+                        }
+                    });
+                }
+                return STATUS_OK;
+            }
+        } catch (Exception e) {
+            return toStatus(e);
+        }
+    }
+
+    @Override
+    public @NonNull UStatus unregisterListener(@NonNull UUri topic, @NonNull UListener listener) {
```

Comment:
Please document API

Reply:
Javadoc should be derived automatically from the interface (SDK). So, we won't duplicate or contradict.